### ISSUE TITLE
Release 1.5.0 breaks rails app boot with eager-loading

### ISSUE THREAD

**Issue by timon** [dry-rb/dry-struct#183]
## Describe the bug

We have noticed the following failure in dependabot PR that bumps version of `dry-struct` to 1.5.0:

```
Run bundle exec rspec spec

An error occurred while loading ./spec/controllers/api/v1/creditsafe/search_spec.rb.
Failure/Error: require File.expand_path('../config/environment', __dir__)
NameError:
  uninitialized constant Dry::Struct::Extensions

# ./vendor/bundle/ruby/3.0.0/gems/dry-core-0.9.0/lib/dry/core/deprecations.rb:227:in `block (2 levels) in deprecate_constant'
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader/helpers.rb:127:in `const_get'
</summary>
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader/helpers.rb:127:in `cget'
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader.rb:246:in `block (2 levels) in eager_load'
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader/helpers.rb:41:in `block in ls'
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader/helpers.rb:27:in `each'
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader/helpers.rb:27:in `ls'
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader.rb:234:in `block in eager_load'
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader.rb:219:in `synchronize'
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader.rb:219:in `eager_load'
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader.rb:318:in `each'
# ./vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.1/lib/zeitwerk/loader.rb:318:in `eager_load_all'
# ./vendor/bundle/ruby/3.0.0/gems/railties-7.0.4/lib/rails/application/finisher.rb:74:in `block in <module:Finisher>'
# ./vendor/bundle/ruby/3.0.0/gems/railties-7.0.4/lib/rails/initializable.rb:32:in `instance_exec'
# ./vendor/bundle/ruby/3.0.0/gems/railties-7.0.4/lib/rails/initializable.rb:32:in `run'
# ./vendor/bundle/ruby/3.0.0/gems/railties-7.0.4/lib/rails/initializable.rb:61:in `block in run_initializers'
# ./vendor/bundle/ruby/3.0.0/gems/railties-7.0.4/lib/rails/initializable.rb:60:in `run_initializers'
# ./vendor/bundle/ruby/3.0.0/gems/railties-7.0.4/lib/rails/application.rb:372:in `initialize!'
# ./config/environment.rb:7:in `<top (required)>'
# ./spec/rails_helper.rb:11:in `require'
# ./spec/rails_helper.rb:11:in `<top (required)>'
# ./spec/controllers/api/v1/creditsafe/search_spec.rb:3:in `require'
# ./spec/controllers/api/v1/creditsafe/search_spec.rb:3:in `<top (required)>'

An error occurred while loading ./spec/controllers/api/v1/documentation_controller_spec.rb.
Failure/Error: require File.expand_path('../config/environment', __dir__)
```

## To Reproduce

A minimalistic rails app that exhibits this behavior is available at https://github.com/timon/dry-failures/pull/1

Please note that

- `CI` should be present at env to enforce eager loading of rails app in a `test` environment
- Just adding `dry-core` seems to work fine

## Expected behavior

Updating to `dry-struct` should not break eager loading of rails apps, neither in CI nor in `production` mode

## My environment

- Affects my production application: **YES**
- Ruby version: 3.0.4
- OS: MacOS, `ubuntu-latest` in github runner

**Comment by texpert** [comment]
I am getting similar `uninitialized constant Dry::Struct::Extensions (NameError)` errors.

**Comment by flash-gordon** [comment]
`Gemfile` doesn't have anything dry-related, what am I missing?

**Comment by timon** [comment]
@flash-gordon I added a link to a PR that demonstrates how adding `dry-struct` to gemfile breaks the CI / app eager loading, please check the commits referenced in https://github.com/timon/dry-failures/pull/1

**Comment by solnic** [comment]
Actually, here's a PR https://github.com/dry-rb/dry-struct/pull/184

Notice that it leads to dry-core but it's just because its deprecations module defines `const_missing` and it's just misled us thinking that it's a problem with dry-core, when in reality it was because we added `extensions` to ignore list in Zeitwerk config, and that actually messed up auto-loading of `Dry::Core::Extensions` API that `Dry::Struct` is extended by.

**Comment by solnic** [comment]
@timon this is now fixed in 1.5.1. Sorry for the trouble and thanks for the helpful reproduction repo 🙂

**Comment by flash-gordon** [comment]
@timon oops I missed it, just clicker at the repo. Thanks for the repro, it always helps
